traffic_stats/traffic_stats.go

```diff
@@ -243,11 +273,18 @@ func main() {
 		case <-termChan:
 			info("Shutdown Request Received - Sending stored metrics then quitting")
 			for _, val := range Bps {
+				if c != nil {
+					publishToKafka(config, val, c)
+				}
 				sendMetrics(config, val, false)
 			}
+			startShutdown(c)
 			os.Exit(0)
 		case <-tickers.Publish:
 			for key, val := range Bps {
+				if c != nil {
```

Comment:
So, rather than checking if the kafka client isn't nil and if InfluxDB is enabled, I think it would make sense to add some kind of `type DataExporter interface { ExportData(bps influx.BatchPoints) }` of which there are two concrete implementations -- one for InfluxDB and one for Kafka. Then, we would have a `dataExporters := []DataExporter{}` which we would add the InfluxDB and Kafka instances to (if enabled). Then, the main goroutine would iterate through the `[]DataExporter` and call `ExportData(...)` on each one. This way, we don't have to constantly check all over the code if certain data exporters are enabled or not.

That said, for `calcDailySummary()`, if that's just going to be an influxDB thing only, then it might make sense to still check if influxDB is enabled in that function.

In the future, if someone wanted to add another data exporter implementation (e.g. Prometheus), this would make it a lot easier. What do you think of this idea?

Reply:
I created a DataExporter interface with a couple caveats: 

1. The retry (bool) parameter is only used for Influx
2. I used the startup config object to implement the method for ExportData (line 152) for Influx (line 284). I wasn't sure what else to use since nothing was needed.